packages/itwinui-react/src/core/ColorPicker/ColorInputPanel.tsx

```diff
@@ -475,10 +509,15 @@ export const ColorInputPanel = React.forwardRef((props, ref) => {
           </IconButton>
         )}
         <Box
-          ref={inputsContainerRef}
-          className='iui-color-input-fields'
+          as='div'
+          {...inputFieldProps}
+          ref={useMergedRefs(inputsContainerRef, inputFieldProps?.ref)}
+          className={cx('iui-color-input-fields', inputFieldProps?.className)}
           role={currentFormat !== 'hex' ? 'group' : undefined}
-          aria-labelledby={currentFormat !== 'hex' ? labelId : undefined}
+          aria-labelledby={
+            inputFieldProps?.id ??
```

Comment:
I'm not sure that this is correct. Shouldn't it be labelled by `panelLabelProps?.id`? And shouldn't it be conditionally labeled only when `currentFormat !== 'hex'`?

It might be better to update the existing `labelId` variable to avoid these mistakes.

Reply:
In terms of updating the existing `labelId`, should it be `panelLabelProps?.id ?? useId()`?

Reply:
yes, except hooks cannot be called conditionally, so you'll have to do something like:

```js
const fallbackLabelId = useId();
const labelid = panelLabelProps?.id ?? fallbackLabelId;
```

Reply:
I have updated `labelId` likewise.